### ISSUE TITLE
osquery support for 32-bit ARM

### ISSUE THREAD

**Issue by thamht4190** [osquery/osquery#6573]
# Blueprint

Hi,

I was trying to run `osquery` on Raspberry Pi 4B. Its architecture is ARM and its recommended OS is 32 bit. I've read [this ](https://github.com/osquery/osquery/issues/1424) and know that your team don't have time to support it. However, can you please give some guideline or idea how to start it? I will try and if there is any difficulty, we can discuss.

Thank you in advance!

**Comment by thamht4190** [comment]
After looking at 2 merged pull requests in https://github.com/osquery/osquery/issues/6285, I don't think I have enough understanding of osquery and osquery-toolchain to do this :( Do you have any plan for it in the near future?

**Comment by mike-myers-tob** [comment]
> After looking at 2 merged pull requests in #6285, I don't think I have enough understanding of osquery and osquery-toolchain to do this :( Do you have any plan for it in the near future?

It's not a priority for anyone at the moment, but if someone wants to work on it, I think the contribution would be welcomed. But I expect there are multiple challenges, like finding a way to automate builds in the CI.

**Comment by theopolis** [comment]
Yes, let's wait until the initial ARM64 patch is merged before starting any work on a 32bit version.

**Comment by kevensen** [comment]
Hi all.  I am curious if there is an update on this effort.

**Comment by theopolis** [comment]
I’m not aware of anyone working on this.

**Comment by knightian** [comment]
Hi, does Apple moving to ARM CPUs have any bearing on this?

**Comment by directionless** [comment]
arm64 is reasonably supported. I don't think arm32 is on anyone's roadmap.

If someone steps up to do the work, I think we'd review the PR. But the big question for new platforms is always around CI and ongoing support burden.
